### ISSUE TITLE
Ninja Shoes, Gloves, and Hood Disappear Randomly

### ISSUE THREAD

**Issue by KomradeSpectre** [tgstation/tgstation#46056]
Issue reported from  Round ID: 116638 (/tg/Station Bagil [ENGLISH] [US-WEST] [100% FREE LAG])

Reporting client version: 512.1477

Right after the ninja of the round was killed, I had taken the leftovers from the place their body blew up, it was a hood, the mask, gloves, and the shoes (the captain took the energy katana). For a while I was running around the station enjoying the speed of the ninja shoes when suddenly *poof* around 2 minutes after that ninja's death, my shoes are gone, my gloves are gone, and my hood is gone, but oddly enough the mask remained. I was greatly disappointed.

**Comment by ATH1909** [comment]
This happened to me too; I didn't report it because I had thought that it was admin intervention or something.

**Comment by nemvar** [comment]
I kinda know why this happens, but diving into ninja code is like diving into hell. The guy who wrote it really doesn't know how to clear references. This probably happens because the suit got qdeleted and took all the other important ninja bits with it.
You aren't even supposed to be able to use the ninja gloves/shoes without the suit but since dismemberment exists and it wasn't accounted for, it's just kinda a thing.

tl;dr ninja is shit gameplay and shit code. Luckily it is modular so anyone can easily remove it at their leasure.

**Comment by IndieanaJones** [comment]
This specific bug has been fixed as of the space ninja rework.
